rust/tools/authorization-logic/src/compilation_top_level.rs

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021 The Raksha Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use crate::{
+    ast::*,
+    parsing::astconstructionvisitor,
+    signing::{export_assertions, import_assertions},
+    souffle::souffle_interface,
+};
+
+use std::fs;
+
+/// The file `compilation_top_level` has the top level function, `compile`
```

Comment:
I think these comments are interpreted as documenting the `source_file_to_ast` function. For a crate-level doc comment, `//!` is the marker to use. see https://doc.rust-lang.org/reference/comments.html#doc-comments

Reply:
Ah thanks I did not know. Applied suggestion broadly.